### ISSUE TITLE
Missing principalId and TenantId property in Azure DataFactory

### ISSUE THREAD

**Issue by adriano-kaiser** [hashicorp/terraform-cdk#587]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

cdktf: 0.10
typescript: 4.1.5
terraform: 0.14.7
azurerm 2.0.0

### Affected Resource(s)

Azure DataFactory

### Expected Behavior

I am trying to create an Azure DataFactory with SystemAssigned Identity and assign that Identity to a new RoleAssignment.
However the object 'principal_id' does not exist in the identity property of the DataFactory Class.

See:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory

> The identity block exports the following:

    principal_id - The ID of the Principal (Client) in Azure Active Directory

    tenant_id - The ID of the Azure Active Directory Tenant.

### Actual Behavior

The properties principal_id and client_id should be added in the DataFactory Class so they can be referenced later, like for example in a role assignment.

```
    const dataFactory = new DataFactory(this, terraformId, {
      name: `mydatafactory`,
      location: config.region,
      resourceGroupName: resourceGroupName,
      identity: [{ type: `SystemAssigned` }],
    });

    new RoleAssignment(scope, `storage-datafactory-assignment`, {
      principalId: dataFactory.identity.principal_id ,
      scope: dataFactoryStorage.id,
      roleDefinitionName: "Storage Blob Data Contributor",
    });
```

So basically this does not exist:
```
dataFactory.identity.principal_id
```

**Comment by jsteinich** [comment]
Sounds like a duplicate of #172.
A fix is in progress; however, it is taking a bit to figure out the best way to add without adding confusion to the creation side.

**Comment by adriano-kaiser** [comment]
Sounds indeed very much like a duplicate.

It did tried using the escape hatch, however its not working, can anyone give me a hint?

```
    var assignment = new RoleAssignment(this, `storage-datafactory-blob-data-contributor-${resourceGroupName}`, {
      principalId: '',
      scope: dataFactoryStorage.id,
      roleDefinitionName: 'Storage Blob Data Contributor',
    });

    assignment.addOverride('principal_id', `${dataFactory.fqn}.identity[0].principal_id`);
```

I am not sure if its correct to leave the principalId empty and assign it later via override, since the principalId is a mandatory property.

`Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidPrincipalId" Message="The Principal ID 'azurerm_data_factory.mydatafactory.identity[0].principal_id' is not valid. Principal ID must be a GUID."`

**Comment by jsteinich** [comment]
Adding as an override afterwards is fine.
Try
```typescript
assignment.addOverride('principal_id', `\${${dataFactory.fqn}.identity[0].principal_id}`);
```

**Comment by adriano-kaiser** [comment]
Thank you so much! That syntax solved my issue!

**Comment by skorfmann** [comment]
Thanks for reporting this issue @adriano-kaiser. Since the suggested workaround by @jsteinich seems to work for you, I'm going to close this issue as a duplicate of https://github.com/hashicorp/terraform-cdk/issues/172

If there's anything else blocking you, please don't hesitate to comment here or open another issue.
